### PR TITLE
Refactor component styles with CSS modules

### PR DESCRIPTION
**Pull request by hugoxiong5**
Add CSS modules and refactor each React component to use modules. 

I meant to try to use composition to share and extend classes between different modules (like with the "modal" class), but I couldn't figure out how to do this. Will return to this later.

### CODE REVIEW

File: client/App.jsx [hunk 1]
```diff
@@ -45,14 +46,29 @@ class App extends React.Component {
...
+    const photosList = this.state.photos.map((photo, index) => (
+      <PhotoCard
+        url={photo}
+        key={index}
+        index={index}
+        clickPhotoOpen={this.clickPhotoOpen}
+      />
+    ));
+
+    const { modalShow, photos, modalScroll } = this.state;
```
Comment (anabellchan) [inline client/App.jsx]: Nice destructuring 😃

File: client/App.jsx [hunk 2]
```diff
@@ -45,14 +46,29 @@ class App extends React.Component {
...
 
     return (
       <div id="photo-gallery">
-        <PhotoModal show={this.state.modalShow} photos={this.state.photos} scroll={this.state.modalScroll} hide={this.hideModal} />
+        <PhotoModal
```
Comment (anabellchan) [inline client/App.jsx]: I'm pretty sure you'll be cleaning this up on later commits.

**Comment by hugoxiong5** [comment]
Also, CSS Modules broke all the client tests, because the client tests reply on finding components by class name selectors. I'll have to fix this at a later point, but I need to ask a SEIR/TM about the best approach.

**Review by anabellchan — APPROVED** [review]
PRs should be focused on a feature, and should not be a subset or superset of another.  
If you're going to change add a new PR with everything thrown in, might as well close the smaller PRs.

File: client/App.jsx (hunk 1 above)
Comment (hugoxiong5) [reply client/App.jsx]: > PRs should be focused on a feature, and should not be a subset or superset of another.
> If you're going to change add a new PR with everything thrown in, might as well close the smaller PRs.

I agree, good point. In a situation where my new PRs depend on my old PRs (which will likely be most of the time), I will close the old PRs. Unfortunately, the new combined PR will be pretty big.

File: client/App.jsx (hunk 2 above)
Comment (hugoxiong5) [reply client/App.jsx]: That's right, I think I changed it to use destructuring.

**Comment by hugoxiong5** [comment]
@anabellchan Thank you for all the hard working reviewing so many files! Much appreciated!

**Pull request merged**